src/main/java/no/unit/nva/amazon/s3/CreateUploadHandler.java

```diff
@@ -0,0 +1,164 @@
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.apache.http.HttpStatus.SC_CREATED;
+import static org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+
+public class CreateUploadHandler implements RequestHandler<Map<String, Object>, GatewayResponse> {
+
+
+    public static final String PARAMETER_BODY_KEY = "body";
+    public static final String PARAMETER_FILENAME_KEY = "filename";
+    public static final String PARAMETER_INPUT_KEY = "input";
+
+    private static final AmazonS3 s3Client = createAmazonS3Client();
+    public final transient String bucketName;
+    private final transient String allowedOrigin;
+
+
+    public CreateUploadHandler() {
+        this(new Environment());
+    }
+
+
+    /**
+     * Construct for lambda eventhandler to create an upload request for S3.
+     */
+    public CreateUploadHandler(Environment environment) {
+        this.allowedOrigin = environment.get(ALLOWED_ORIGIN_KEY)
+                .orElseThrow(() -> new  IllegalStateException(String.format(MISSING_ENV_TEXT,ALLOWED_ORIGIN_KEY)));
+        this.bucketName = environment.get(S3_UPLOAD_BUCKET_KEY)
+                .orElseThrow(() -> new  IllegalStateException(String.format(MISSING_ENV_TEXT,S3_UPLOAD_BUCKET_KEY)));
+
+
+    }
+
+    public static AmazonS3 createAmazonS3Client() {
+        return AmazonS3ClientBuilder.standard()
+                .build();
+    }
+
+    public AmazonS3 getS3Client() {
+        return s3Client;
+    }
+
+    @Override
+    public GatewayResponse handleRequest(Map<String, Object> input, Context context) {
+
+        final GatewayResponse response = new GatewayResponse();
+        response.setHeaders(headers());
+
+
+        CreateUploadRequestBody requestBody = null;
+        try {
+            requestBody = checkParameters(input);
+        } catch (JsonSyntaxException | ParameterMissingException e) {
+            System.out.println(DebugUtils.dumpException(e));
+            response.setErrorBody(e.getMessage());
+            response.setStatusCode(SC_BAD_REQUEST);
+            System.out.println(response);
+            return response;
+        } catch (Exception e) {
+            System.out.println(DebugUtils.dumpException(e));
+            response.setErrorBody(e.getMessage());
+            response.setStatusCode(SC_INTERNAL_SERVER_ERROR);
+            System.out.println(response);
+            return response;
+        }
+
+
+        try {
+            ObjectMetadata objectMetadata = getObjectMetadata(requestBody);
+
+            String keyName = UUID.randomUUID().toString();
+            InitiateMultipartUploadRequest initRequest =
+                    new InitiateMultipartUploadRequest(bucketName, keyName, objectMetadata);
+            InitiateMultipartUploadResult initResponse = getS3Client().initiateMultipartUpload(initRequest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code manually manages uploads or downloads. The SDK can automatically handle these operations. Consider using the SDK to automatically handle this instead. SDK class: TransferManager. See [Using TransferManager for Amazon S3 Operations](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/examples-s3-transfermanager.html).